src/runtimes/node/bundlers/nft/transpile.ts

```diff
@@ -23,8 +24,44 @@ export const transpile = async ({ config, format, name, path }: TranspileOptions
     const transpiled = await build({
       bundle: false,
       entryPoints: [path],
+      format: MODULE_FORMAT.COMMONJS,
+      logLevel: 'error',
+      platform: 'node',
+      sourcemap: Boolean(config.nodeSourcemap),
+      target: [nodeTarget],
+      tsconfigRaw: tsConfig ? JSON.stringify(tsConfig) : undefined,
+      write: false,
+    })
+
+    return transpiled.outputFiles[0].text
+  } catch (error) {
+    throw FunctionBundlingUserError.addCustomErrorInfo(error, {
+      functionName: name,
+      runtime: RUNTIME.JAVASCRIPT,
+      bundler: NODE_BUNDLER.NFT,
+    })
+  }
+}
+
+interface TranspileTSOptions {
+  config: FunctionConfig
+  format?: ModuleFormat
+  name: string
+  path: string
+}
+
+export const transpileTS = async ({ config, format, name, path }: TranspileTSOptions) => {
+  // The version of ECMAScript to use as the build target. This will determine
+  // whether certain features are transpiled down or left untransformed.
+  const nodeTarget = getBundlerTarget(config.nodeVersion)
+
+  try {
+    const transpiled = await build({
+      bundle: true,
```

Comment:
does this mean we are now bundling non v2 as well? or should we gate the bundling to only the v2 interface to not break old and existing sites without opting into the new interface?

Reply:
Done in https://github.com/netlify/zip-it-and-ship-it/pull/1584/commits/215595ac0e1337ed8e585fdf5e61c60a851d74a0.